### PR TITLE
[Feature][1.3.1] linkis linkis-engineconn-plugins add seatunnel

### LINKED ISSUE #3260: [Feature][linkis-engineconn-plugins]  linkis-engineplugin-seatunnel
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-linkis/issues) and found no similar feature requirement.

### Problem Description

SeaTunnel is a very easy-to-use ultra-high-performance distributed data integration platform that supports real-time synchronization of massive data. It can synchronize tens of billions of data stably and efficiently every day, and has been used in the production of nearly 100 companies.
1.linkis new seatunnel engine

### Description

_No response_

### Use case

_No response_

### solutions

_No response_

### Anything else

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by dlimeng**
### What is the purpose of the change
Extend linkis  seatunnel
Fixes  #3260

### Brief change log
linkis-engineconn-plugins add seatunne

### Does this pull request potentially affect one of the following parts:
1.linkis-engineconn-plugins add seatunne

### CODE REVIEW

File: linkis-engineconn-plugins/seatunnel/src/main/java/org/apache/linkis/engineconnplugin/seatunnel/client/utils/JarLoader.java [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+import java.net.URLClassLoader;
+import java.security.AccessControlContext;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class JarLoader extends URLClassLoader {
+  private AccessControlContext acc;
```
Comment (casionone) [inline linkis-engineconn-plugins/seatunnel/src/main/java/org/apache/linkis/engineconnplugin/seatunnel/client/utils/JarLoader.java:34]: The sqoop engine also with the same java class org.apache.linkis.engineconnplugin.sqoop.client.utils.JarLoader 
move to the common module and reuse it?

File: linkis-engineconn-plugins/seatunnel/src/main/scala/org/apache/linkis/engineconnplugin/seatunnel/SeatunnelEngineConnPlugin.scala [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+class SeatunnelEngineConnPlugin extends EngineConnPlugin {
+  private val EP_CONTEXT_CONSTRUCTOR_LOCK = new Object()
+  private var engineResourceFactory: EngineResourceFactory = _
+  private var engineConnLaunchBuilder: EngineConnLaunchBuilder = _
+  private var engineConnFactory: EngineConnFactory = _
+  override def init(params: java.util.Map[String, Any]): Unit = {}
+
+  override def getEngineResourceFactory: EngineResourceFactory = {
+
+    EP_CONTEXT_CONSTRUCTOR_LOCK.synchronized {
```
Comment (peacewong) [inline linkis-engineconn-plugins/seatunnel/src/main/scala/org/apache/linkis/engineconnplugin/seatunnel/SeatunnelEngineConnPlugin.scala:40]: It is recommended to use a different lock object

File: linkis-engineconn-plugins/seatunnel/src/main/scala/org/apache/linkis/engineconnplugin/seatunnel/util/SeatunnelUtils.scala [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+import org.apache.linkis.engineconnplugin.seatunnel.config.SeatunnelSparkEnvConfiguration
+
+import org.apache.commons.io.IOUtils
+import org.apache.commons.logging.Log
+import org.apache.commons.logging.LogFactory
+
+import java.io.{BufferedReader, File, InputStreamReader, PrintWriter}
+
+object SeatunnelUtils {
+  val logger: Log = LogFactory.getLog(getClass)
```
Comment (peacewong) [inline linkis-engineconn-plugins/seatunnel/src/main/scala/org/apache/linkis/engineconnplugin/seatunnel/util/SeatunnelUtils.scala:31]: slf4j is recommended

File: linkis-engineconn-plugins/seatunnel/src/main/scala/org/apache/linkis/engineconnplugin/seatunnel/util/SeatunnelUtils.scala [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+  }
+
+  def executeLine(code: String): Int = {
+    var bufferedReader: BufferedReader = null
+    var errorsReader: BufferedReader = null
+    try {
+      val processBuilder: ProcessBuilder = new ProcessBuilder(generateRunCode(code): _*)
+      process = processBuilder.start()
+      bufferedReader = new BufferedReader(new InputStreamReader(process.getInputStream))
+      errorsReader = new BufferedReader(new InputStreamReader(process.getErrorStream))
```
Comment (peacewong) [inline linkis-engineconn-plugins/seatunnel/src/main/scala/org/apache/linkis/engineconnplugin/seatunnel/util/SeatunnelUtils.scala:56]: error does not seem to be used, it is recommended to refer to org.apache.linkis.manager.engineplugin.shell.executor.ReaderThread of shell ec

File: linkis-engineconn-plugins/seatunnel/src/main/java/org/apache/linkis/engineconnplugin/seatunnel/client/utils/JarLoader.java (hunk 1 above)
Comment (Dlimeng) [reply linkis-engineconn-plugins/seatunnel/src/main/java/org/apache/linkis/engineconnplugin/seatunnel/client/utils/JarLoader.java:34]: 1

Comment (Dlimeng) [reply linkis-engineconn-plugins/seatunnel/src/main/java/org/apache/linkis/engineconnplugin/seatunnel/client/utils/JarLoader.java:34]: ok

File: linkis-computation-governance/linkis-manager/linkis-label-common/src/main/java/org/apache/linkis/manager/label/conf/LabelCommonConfig.java [hunk 5]
```diff
@@ -89,4 +89,7 @@ public class LabelCommonConfig {
 
   public static final CommonVars<String> ELASTICSEARCH_ENGINE_VERSION =
       CommonVars.apply("wds.linkis.elasticsearch.engine.version", "7.6.2");
+
+  public static final CommonVars<String> SEATUNNEL_ENGINE_CONN_VERSION =
+      CommonVars.apply("wds.linkis.seatunnel.engineconn.version", "2.1.2");
```
[the remark this answers is not in the document]
Comment (Dlimeng) [reply linkis-computation-governance/linkis-manager/linkis-label-common/src/main/java/org/apache/linkis/manager/label/conf/LabelCommonConfig.java:94]: ok

File: linkis-engineconn-plugins/seatunnel/pom.xml [hunk 6]
```diff
@@ -0,0 +1,267 @@
...
+  <parent>
+    <groupId>org.apache.linkis</groupId>
+    <artifactId>linkis</artifactId>
+    <version>1.3.0</version>
+    <relativePath>../../pom.xml</relativePath>
+  </parent>
+
+  <artifactId>linkis-engineplugin-seatunnel</artifactId>
+  <properties>
+    <maven.compiler.source>8</maven.compiler.source>
```
[the remark this answers is not in the document]
Comment (Dlimeng) [reply linkis-engineconn-plugins/seatunnel/pom.xml:29]: ok

**Review by jacktao007 — APPROVED** [review]
LGTM.

File: linkis-engineconn-plugins/seatunnel/src/main/java/org/apache/seatunnel/core/base/config/ConfigBuilder.java [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+import java.nio.file.Path;
+
+/**
+ * Used to build the {@link Config} from file.
+ *
+ * @param <ENVIRONMENT> environment type.
+ */
+public class ConfigBuilder<ENVIRONMENT extends RuntimeEnv> {
+
+  private static final Log LOGGER = LogFactory.getLog(SparkStarter.class);
```
Comment (casionone) [inline linkis-engineconn-plugins/seatunnel/src/main/java/org/apache/seatunnel/core/base/config/ConfigBuilder.java:39]: recommend to use slf4j

**Comment by dlimeng** [comment]
> ```
> seatunnel-core-flink-2.1.2.jar
> seatunnel-core-flink-sql-2.1.2.jar
> seatunnel-core-spark-2.1.2.jar
> clickhouse-jdbc-0.3.2-patch11.jar
> postgresql-42.3.1.jar 
> ```
> 
> this new dependencies license need add to linkis-dist/release-docs/xxxx

ok

File: linkis-engineconn-plugins/seatunnel/src/main/java/org/apache/seatunnel/core/base/config/ConfigBuilder.java (hunk 7 above)
Comment (dlimeng) [reply linkis-engineconn-plugins/seatunnel/src/main/java/org/apache/seatunnel/core/base/config/ConfigBuilder.java:39]: ok

File: linkis-engineconn-plugins/seatunnel/src/main/java/org/apache/seatunnel/core/base/config/PluginFactory.java [hunk 8]
```diff
@@ -0,0 +1,238 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.seatunnel.core.base.config;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
```
[the remark this answers is not in the document]
Comment (dlimeng) [reply linkis-engineconn-plugins/seatunnel/src/main/java/org/apache/seatunnel/core/base/config/PluginFactory.java:22]: ok

**Review by casionone — APPROVED** [review]

**Pull request merged**